content/blog/new-graphic-features/index.Rmd

```diff
@@ -0,0 +1,301 @@
+
+```{r, eval = FALSE}
+install.packages("ragg")
+install.packages("svglite")
+```
+
+The post will rarely make any specific call-outs to ragg or svglite, as these are simply the packages that facilitate what is now possible with R graphics.
+
+## What is the graphics engine?
+
+You might wonder what is meant by the *R graphics engine*. It's pretty deep in the R graphics stack, so as a user, you are unlikely to ever engage with it directly, but since I somehow caught your attention we might as well indulge in the finer points of the graphics implementation.
+
+While you may mainly be familiar with ggplot2 and perhaps a variety of graphics devices (e.g. `pdf()` or `png()`), they sit at opposite ends of a fairly elaborate graphics pipeline. ggplot2 is a high(er) level plotting package that allows you to express your data visualization intent through a structured grammar. Graphics devices such as ragg and svglite are low-level packages that translate simple graphics instructions into a given file format. In between these two poles we have a two additional abstractions that helps translate between the extremes. In very broad terms, the R graphic stack looks like this:
+
+![An overview of the different steps in the R graphics pipeline. A graphics package is build on top of a graphic system. All graphic systems calls into the same shared graphics engine which then relay graphic instructions to the active graphic device.](pipeline.png "graphics pipeline")
+
+### Graphic systems
+
+R currently sports two different systems, one colloquially known as *base* graphics (implemented in the graphics package), and one called *grid* graphics (implemented in the grid package). If you call `plot()` you are most likely to end up using the base graphics system, while grid is used as the basis for e.g. ggplot2. The two systems are largely silos, though effort has been made to allow users to embed base graphics into grid graphics. In RStudio we are mainly invested in the grid graphics system since it powers ggplot2, but by and large, this is all an implementation detail that the user shouldn't care too much about. There might come other graphic systems in the future, and other ways of drawing things on screen or to files also exist outside of the R graphics pipeline (e.g. rgl which allows you to create 3D graphics using an OpenGL/WebGL interface).
+
+### The graphics engine
+
+What unites base and grid graphics and sets them apart from e.g. rgl is that they both call into the same low level C API provided by R, the **graphics engine**. The graphics engine is responsible for communicating with the graphics devices while also providing a selected range of utility functionality common for both base and grid graphics. It is because of this abstraction that graphics in R is largely decoupled from how it is outputted, be it on screen, in a file, or directly to a printer.
```

Comment:
```suggestion
What unites base and grid graphics and sets them apart from e.g. rgl is that they both call into the same low-level C API provided by R, the **graphics engine**. The graphics engine is responsible for communicating with the graphics devices while also providing selected utility functionality common for both base and grid graphics. It is because of this abstraction that graphics in R is largely decoupled from how it is outputted, be it on screen, in a file, or directly to a printer.
```

Reply:
> It is because of this abstraction that graphics in R is largely decoupled…

It feels like you're missing a word after graphics here. Graphics generation? (IDK the right word, since it's probably quite technical). If you're _not_ missing a word, then it should be "graphics in R _are_ largely decoupled…"